app/core/auth.py

```
@@ -0,0 +1,52 @@
+        # 토큰 디코딩 및 검증
+        payload = jwt.decode(token, base64.b64decode(settings.JWT_SECRET_KEY), algorithms=[settings.JWT_ALGORITHM])
+
+        user_id_str: str = payload.get("userId")
+        if user_id_str is None:
+            raise credentials_exception
+
+        return int(user_id_str)
+
+    except JWTError as e:
+        print(f"\n[DEBUG] JWTError 발생: {e}")
+        raise credentials_exception
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)